src/dfm/common/data_cleaning/dolma_taggers/language_scandi.py

```diff
@@ -0,0 +1,202 @@
+class Cld2LanguageFilterScandi(BaseTagger):
+    RE_BAD_CHARS = regex.compile(r"[\p{Cc}\p{Cs}]+")
+
+    def _sanitize_input(self, text: str) -> str:
+        return self.RE_BAD_CHARS.sub("", text)
+
+    def _to_ascii_input(self, text: str) -> str:
+        return anyascii(text)
+
+    def _identity_fn(self, text: str) -> str:
+        return text
+
+    def _predict_text(self, text: str) -> dict[str, float]:
+        is_reliable = False
+        details: Iterable[tuple[str, str, int, float]] = []
+        for fn in (self._identity_fn, self._to_ascii_input, self._sanitize_input):
+            try:
+                retvals = cld2.detect(fn(text))
+                assert len(retvals) == 3
+                is_reliable, _, details = retvals
+                break
+            except cld2.error:
+                ...
+
+        scores: dict[str, float] = {}
+        if is_reliable:
+            for lang, _, score, _ in details:
+                if lang in LANGS:
+                    scores[LANGS[lang]] = score / 100.0
+
+        return scores
+
+    def predict(self, doc: Document) -> DocResult:
+        lang_scores = self._predict_text(doc.text)
+        spans: list[Span] = []
+        for lang_code in LANGS.values():
+            score = lang_scores.get(lang_code, 0)
+
+            positive_span = Span(
+                start=0,
+                end=len(doc.text),
+                type=lang_code,
+                score=score,
+            )
+            negative_span = Span(
+                start=0,
+                end=len(doc.text),
+                type=f"not_{lang_code}",
+                score=1.0 - score,
+            )
+            spans.append(positive_span)
+            spans.append(negative_span)
+        return DocResult(doc=doc, spans=spans)
+
+
+@TaggerRegistry.add("cld2_scandi_paragraph")
+class Cld2LanguageFilterParagraphScandi(Cld2LanguageFilterScandi):
+    def predict(self, doc: Document) -> DocResult:
+        paragraphs = split_paragraphs(doc.text)
+        spans: list[Span] = []
+        for paragraph in paragraphs:
+            lang_scores = self._predict_text(paragraph.text)
+            for lang_code in LANGS.values():
+                score = lang_scores.get(lang_code, 0.0)
+
+                positive_span = Span(
+                    start=paragraph.start,
+                    end=paragraph.end,
+                    type=lang_code,
+                    score=score,
+                )
+                negative_span = Span(
+                    start=paragraph.start,
+                    end=paragraph.end,
+                    type=f"not_{lang_code}",
+                    score=1.0 - score,
+                )
+                spans.extend((positive_span, negative_span))
+        return DocResult(doc=doc, spans=spans)
+
+
+@TaggerRegistry.add("ft_lang_id_scandi_doc")
+class FastTextScandiLanguageDocumentTagger(BaseFastTextTagger):
+    MODEL_PATH = "https://dl.fbaipublicfiles.com/fasttext/supervised-models/lid.176.bin"
```

Comment:
Can we add a reference to the models (potentially in the docstring), would love to check the performance metrics.